### PR TITLE
Remove tensor option for _global_exception_occured

### PR DESCRIPTION
**Pull request by irenedea**
# What does this PR do?

Makes it able to pass MLFlowLogger to a child process with `export PYTORCH_CUDA_ALLOC_CONF="expandable_segments:True"`

Addresses the error:
```
RuntimeError: Tensors allocated with expandable_segments:True cannot be shared between processes. Consider using expandable_segments:False in data loading workers via torch.cuda.memory._set_allocator_settings('expandable_segments:False')
```

Run with fix: test-hf-checkpoint-fix-Vknjzg
Run without fix: test-hf-checkpoint-baseline-5ZGdqG

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/main/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/main/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/main/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/loggers/mlflow_logger.py [hunk 1]
```diff
@@ -615,10 +612,7 @@ def post_close(self):
             if hasattr(self, 'monitor_process'):
                 # Check if there is an uncaught exception, which means `post_close()` is triggered
                 # due to program crash.
-                if isinstance(self._global_exception_occurred, torch.Tensor):
-                    finish_with_exception = (self._global_exception_occurred == 1).item()
-                else:
-                    finish_with_exception = (self._global_exception_occurred == 1)
+                finish_with_exception = (self._global_exception_occurred == 1)
```
Comment (mvpatel2000) [inline composer/loggers/mlflow_logger.py:615]:
```suggestion
                finish_with_exception = self._global_exception_occurred == 1
```

**Review by mvpatel2000 — COMMENTED** [review]
nit

**Review by chenmoneygithub — APPROVED** [review]
LGTM!

**Review by mvpatel2000 — APPROVED** [review]
LGTM

**Pull request merged**